### PR TITLE
Disable calving into holes in ice shelf

### PR DESCRIPTION
**Pull request by matthewhoffman**
This merge adds a mask for open ocean that is seeded by open ocean cells along the mesh boundary and then flood-filled inward.  This mask is then used for identifying edges where calving should be applied to disable calving into 'holes' in ice shelves that can form from localized melting or other unusual situations.

### CODE REVIEW

**Comment by matthewhoffman** [comment]
@trhille , here is the ice-shelf hole PR.  I will run the integration suite and demonstrate its behavior next week, but I figured I'd get it on your radar now.

File: components/mpas-albany-landice/src/mode_forward/mpas_li_calving.F [hunk 1]
```diff
@@ -3011,6 +3026,92 @@ function scale_edge_length(angleEdgeHere, u, v)
...
+               if (jCell == nCells + 1) then
+                  seedMask(iCell) = 1  ! this is the seed locations - open ocean along domain boundary
+                  exit
+               endif
+            enddo
+         endif
+      enddo
+
+      call mpas_timer_start("halo updates")
+      call mpas_dmpar_field_halo_exch(domain, 'seedMask')
```
Comment (trhille) [inline components/mpas-albany-landice/src/mode_forward/mpas_li_calving.F:3104]: Is this not necessary for `growMask` as well?

**Comment by trhille** [comment]
All `full_integration` tests pass baseline comparison.

**Comment by trhille** [comment]
@matthewhoffman, this is looking good! There might be one little thing to tweak, but I'm not sure where exactly the issue is arrising. To test this, I took a 2050 restart file from one of my Thwaites runs and cut out a hole in the ice shelf:
image

I then ran it forward for five years with the velocity solver turned off. The hole mostly does not calve, but there are a few cells on the edge of the hole  in most time steps that have calving thickness of a few meters. For example:
image

Maybe we need a halo update somewhere? Or maybe there is something about non-dynamic cells that we need to check?

**Comment by trhille** [comment]
It occurred to me that the small bits of calving I showed above are icebergs that are getting cleaned up by the `remove_icebergs` routine, which is obviously fine. I'll try a run what that turned off now.

**Comment by trhille** [comment]
I've confirmed that there is never `frontAblationMask` > 0 on the edges of the hole, which demonstrates correct behavior:
image

**Comment by trhille** [comment]
Oh right! I had assumed that all those clean-up loops used `frontAblationMask`. I guess I should have looked closer :)

**Review by trhille — APPROVED** [review]
Thanks, @matthewhoffman!

**Pull request merged**